Message-ID: <496C6C6A.1050407@mail.mipt.ru>
Date: Tue, 13 Jan 2009 13:26:50 +0300
From: Sergey Ovcharenko <velegrin@mail.mipt.ru>
User-Agent: Thunderbird 2.0.0.14 (X11/20080603)
MIME-Version: 1.0
To: stern@rowland.harvard.edu, linux-usb@vger.kernel.org,
        usb-storage@lists.one-eyed-alien.net

Petr Lanc reported problems with his nokia 6300 in mass-storage mode:
http://bugs.gentoo.org/show_bug.cgi?id=253034

The number of reported sectors is off-by-one.
This bug used to be fixed,but the devices revision 
changed to 7.00

Signed-off-by: Sergey Ovcharenko <velegrin@mail.mipt.ru>
---
/proc/bus/usb/devices:
T:  Bus=02 Lev=01 Prnt=01 Port=04 Cnt=01 Dev#=  4 Spd=12  MxCh= 0
D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
P:  Vendor=0421 ProdID=04fa Rev= 7.00
S:  Manufacturer=Nokia
S:  Product=Nokia 6300
S:  SerialNumber=356857024913660
C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=100mA
I:* If#= 0 Alt= 0 #EPs= 2 Cls=08(stor.) Sub=06 Prot=50 Driver=usb-storage
E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms
E:  Ad=01(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms


Index: linux-2.6.27-gentoo-r6/drivers/usb/storage/unusual_devs.h
=================================================================== 
--- linux-2.6.27-gentoo-r7/drivers/usb/storage/unusual_devs.h.orig
+++ linux-2.6.27-gentoo-r7/drivers/usb/storage/unusual_devs.h
@@ -247,7 +247,7 @@ UNUSUAL_DEV(  0x0421, 0x04b9, 0x0500, 0x
 		US_FL_FIX_CAPACITY ),
 
 /* Reported by Richard Nauber <RichardNauber@web.de> */
-UNUSUAL_DEV(  0x0421, 0x04fa, 0x0550, 0x0660,
+UNUSUAL_DEV(  0x0421, 0x04fa, 0x0550, 0x0700,
 		"Nokia",
 		"6300",

 		US_SC_DEVICE, US_PR_DEVICE, NULL,





